src/Bicep.Core/Analyzers/Linter/Rules/InterpolateNotConcatRule.cs

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.CodeAction;
+using Bicep.Core.Parsing;
+using Bicep.Core.PrettyPrint;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class InterpolateNotConcatRule : LinterRuleBase
+    {
+        public new const string Code = "prefer-interpolation";
+
+        public InterpolateNotConcatRule() : base(
+            code: Code,
+            description: CoreResources.InterpolateNotConcatRuleDescription,
+            docUri: "https://aka.ms/bicep/linter/prefer-interpolation")
+        { }
+
+        private CodeReplacement GetCodeReplacement(TextSpan span)
```

Comment:
>GetCodeReplacement [](http://example.com/codeflow?start=32&length=18)

Is this actually used?

Reply:
Used by CreateFix which is used by VisitFunctionCallSyntax

Reply:
Are you sure? Those seem to pass in a FunctionCallSyntax rather than a TextSpan. Also, why would we produce a replacement with a suffix " - this is the new code"?

Reply:
@jfleisher Can you comment?

Reply:
not needed